src/screens/PinFlow/CurrentPinScreen.tsx

```diff
@@ -59,11 +120,30 @@ export class CurrentPinScreen extends PureComponent<Props, State> {
           <PinInput value={this.state.pin} onTextChange={this.updatePin} />
           <Text style={styles.errorText}>{error}</Text>
         </View>
+        {this.isTimeCounterVisible() && (
+          <TimeCounter
+            navigation={this.props.navigation}
+            onCountFinish={this.onCountFinish}
+            isVisible={this.isTimeCounterVisible()}
+            timestamp={this.props.timeCounter.timestamp}
+          />
+        )}
       </ScreenTemplate>
     );
   }
 }
 
+const mapStateToProps = (state: ApplicationState) => ({
+  timeCounter: state.timeCounter,
```

Comment:
we have library reselect for getting value from the state, please check for example file state/authenticators/selectors and make something similar

Reply:
@KedziaPawel  `reselect` is used on the develop branch, this ticket branch will be created from master. I could add reselect at `master` branch but what will be the pros of that in that particular case? (I've didn't used `reselect` before)

Reply:
if this code doesn't go to develop you don't have to use reselect